### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by coolreader18**
Includes the wasi fix for the `instant` crate

### CODE REVIEW

File: vm/src/stdlib/os.rs [hunk 1]
```diff
@@ -2080,27 +2080,27 @@ mod posix {
 
     #[pyfunction(name = "WIFSIGNALED")]
     fn wifsignaled(status: i32) -> bool {
-        unsafe { libc::WIFSIGNALED(status) }
+        libc::WIFSIGNALED(status)
```
Comment (youknowone) [inline vm/src/stdlib/os.rs:2083]: what's happening here? is this safe in linux but unsafe in macos?

Comment (coolreader18) [reply vm/src/stdlib/os.rs:2083]: Looks like it, I guess I'll just put `#[allow(unused_unsafe)]`

Comment (youknowone) [reply vm/src/stdlib/os.rs:2083]: https://github.com/rust-lang/libc/issues/1888

Comment (youknowone) [reply vm/src/stdlib/os.rs:2083]: how about fixing libc as 0.2.76 for a while? It seems they will be all safe after fixing the issue

**Comment by youknowone** [comment]
Does CI uses older version of Rust for windows? This is somewhat unexpected.

**Comment by coolreader18** [comment]
I added a rust-toolchain file; I don't think we've ever explicitly talked about what versions we support but I feel like we've never shied away from using features in a version that just released, so I think "latest stable" is a good thing to keep to.

**Review by youknowone — APPROVED** [review]

**Pull request merged**